### ISSUE TITLE
support gh-ost

### ISSUE THREAD

**Issue by allixsenos** [square/shift#43]
gh-ost seems to be a better solution than pt-online-schema, it'd be amazing if it were available as an option

**Comment by michaelfinch** [comment]
Agreed. We plan to add gh-ost integration and let you choose between using that and pt-osc. I'm hoping to have that done by the end of this year.

**Comment by ramkireddy** [comment]
Hi, In your blog post on shift (https://medium.com/square-corner-blog/shift-safe-and-easy-database-migrations-42108ee216f#.yt6jep9x1) you mentioned about following future enhancements, do you have any integration in-place to work with flyway or liquibase database migration tools? Would be interested to know if you have made any progress on this front.

"we plan to build into it in the near future to make it more robust, one example being to make it easier to integrate with developer workflows (ex: direct integration with ActiveRecord)"

Thanks,
Ramki

**Comment by michaelfinch** [comment]
Hey @ramkireddy, haven't made much progress on that front unfortunately.

Regarding gh-ost: I have begun the work to integrate it, but still need to find time to push that to completion. I'll post an update about that here when I have one

**Comment by sahubharat25** [comment]
Hi,
Does shift  take care of replication lag  same like ghost or not.

**Comment by michaelfinch** [comment]
Shift uses pt-online-schema-change, which will monitor slave lag for you (https://www.percona.com/doc/percona-toolkit/2.1/pt-online-schema-change.html#cmdoption-pt-online-schema-change--max-lag)

**Comment by gilahacker** [comment]
Just throwing out a +1 for gh-ost support. Shift looks fantastic, but our devs have made extensive use of triggers, so pt-osc is sadly not an option for us.

**Comment by rbroemeling** [comment]
@michaelfinch Just as a quick query -- has any progress been made on using `gh-ost` instead of `pt-online-schema-change`? Thanks!

**Comment by michaelfinch** [comment]
Unfortunately it has not. It's still on my team's roadmap at Square, but it keeps getting punted on in favor of higher-priority work.

**Comment by mailankitkm** [comment]
@michaelfinch We are waiting for gh-ost support in shift. shift is very useful tool to put a process around schema changes. Please push gh-ost support. Thanks.
